util/topgen/merge.py

```diff
@@ -134,6 +124,54 @@ def elaborate_instance(instance, block: IpBlock):
     # to convert and copy them here.
     instance["inter_signal_list"] = [s.as_dict() for s in block.inter_signals]
 
+    # An instance must either have a 'base_addr' address or a 'base_addrs'
+    # address, but can't have both.
+    base_addrs = instance.get('base_addrs')
+    if base_addrs is None:
```

Comment:
sorry i'm not entirely keeping up with the recent changes, but shouldn't some of the checks shown here be over in validate.py? Or is that somewhat out of date?

Reply:
Hmm, you're right, it could go in either place. Looking at the rest of merge.py, I think that there's already lots of validation going on outside of validate.py. The error recovery story isn't currently great in topgen as a whole: while doing this, I spent a lot of time with it exploding on type errors in unrelated code with no error message! My preference is probably to leave this here, but try to do something like I did for reggen and add proper parsing and classes sometime in the nearish future.